airflow/cli/commands/task_command.py

```diff
@@ -156,8 +156,10 @@ def _get_dag_run(
     raise ValueError(f"unknown create_if_necessary value: {create_if_necessary!r}")
 
 
+@internal_api_call
 @provide_session
-def _get_ti(
+def _get_ti_db_access(
+    dag: DAG,
     task: Operator,
```

Comment:
Maybe should add a check to guard misuses where the task is from a different DAG.

Reply:
ok